internal/vehiclepositions/api.go

```
@@ -56,10 +73,41 @@ func VehiclePositionsHandler(context IConnectors) gin.HandlerFunc {
 	}
 }
 
-func InitVehiclePositionrequestParameter(c *gin.Context) (param *VehiclePositionRequestParameter) {
+func InitVehiclePositionrequestParameter(c *gin.Context) (param *VehiclePositionRequestParameter, err error) {
+	var longitude, latitude float64
+	var e error
 	p := VehiclePositionRequestParameter{}
 	p.VehicleJourneyCodes = c.Request.URL.Query()["vehicle_journey_code[]"]
 
+	distanceStr := c.DefaultQuery("distance", "")
+	p.Distance = utils.StringToInt(distanceStr, 0)
+	coordStr := c.Query("coord")
+	if len(coordStr) > 0 {
+		p.Distance = utils.StringToInt(distanceStr, 500)
+	}
+
+	if p.Distance > 0 && len(p.VehicleJourneyCodes) == 0 {
+		if len(coordStr) == 0 {
+			return nil, fmt.Errorf("Bad request: coord is mandatory")
+		}
+		coord := strings.Split(coordStr, ";")
+		if len(coord) == 2 {
```

If `len(coord) != 2`, shouldn't we return an error code, like in the case where `len(coordStr) == 0` (in both cases, the coordinate is not available, either not valid or not present)?